util/tracing/transform/span.go

```diff
@@ -1,259 +1,205 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package transform
 
 import (
-	"time"
-
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/sdk/instrumentation"
-	"go.opentelemetry.io/otel/sdk/resource"
 	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
-	v11 "go.opentelemetry.io/proto/otlp/common/v1"
-	v1 "go.opentelemetry.io/proto/otlp/resource/v1"
 	tracepb "go.opentelemetry.io/proto/otlp/trace/v1"
 )
 
-const (
-	maxMessageEventsPerSpan = 128
-)
-
-// Spans transforms slice of OTLP ResourceSpan into a slice of SpanSnapshots.
-func Spans(sdl []*tracepb.ResourceSpans) []tracesdk.ReadOnlySpan {
+// Spans transforms a slice of OpenTelemetry spans into a slice of OTLP
+// ResourceSpans.
+func Spans(sdl []tracesdk.ReadOnlySpan) []*tracepb.ResourceSpans {
```

Comment:
The function now does the opposite? 🤔

Reply:
🤔 

Maybe it was replaced with something.

Reply:
@ Aneurysm9 @mralias Any suggestions? This was used to forward the traces from containers/clients to whatever telemetry controller was configured with buildkit daemon, without containers needing to configure telemetry for themselves(they wouldn't have network access to do that anyway).

Reply:
I don't think the stated [source](https://github.com/open-telemetry/opentelemetry-go/blob/cfed99561d50f42d9499e05559ef378d8170d703/exporters/otlp/otlptrace/internal/tracetransform/span.go) of this file was actually the source. Looking through the history of that source it always transformed in the direction it currently does.

I can't say with any certainty, but I expect the original in this project was not copied from that source. I think that direction of the transformation is still be possible, but it would need to be written out instead of generated.

Reply:
@AkihiroSuda I think I might be remembering it wrong as I also can't find the old code in otel repo. So I guess I wrote this package myself as a reverse version of the upstream one. In that case, we should just look at what has changed in the upstream package and if there are any new keys and logic there, and make matching changes to our package so we don't drop any information during the conversion.